### PR TITLE
port network check to golang

### PR DESCRIPTION
**Pull request by arbll**
### What does this PR do?

Port the [network check](https://docs.datadoghq.com/integrations/network/) to golang.

### Motivation

This port will be used in environments where python checks are not shipped with the agent. Regular agent will still use the old python check.

### Additional Notes

- The objective is not to be a perfect drop-in replacement of the python check, just to be close enough so that the public doc is usable for both checks
- A few undocumented (as in not in the config template or on the [network check](https://docs.datadoghq.com/integrations/network/)) options are not configurable:
  - `collect_rate_metrics` (default `true` in py): toggle sending TCP metrics as rate. `true` behavior implemented in go
  - `collect_count_metrics` (default `false` in py): toggle sending TCP metrics as count (same values as above but sent as count with a `.count` suffix). `true` behavior implemented in go
  - `combine_connection_states` (default `true` in py): toggle aggregating tcp states by "groups". `true` behavior implemented in go

- Only linux (and android but it's more or less the same thing) needs to be supported for now. With the current implementation darwin and openbsd might also work.
- I had to rename the `network` package to `net` because the linter was failing on a repetition `network.networkCheck`......

### Testing

Testing progress:
- [x] linux
- [x] android

### CODE REVIEW

File: pkg/collector/corechecks/net/network.go [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+	if err != nil {
+		return err
+	}
+	for _, interfaceIO := range ioByInterface {
+		if !c.isDeviceExcluded(interfaceIO.Name) {
+			submitInterfaceMetrics(sender, interfaceIO)
+		}
+	}
+
+	protocols := []string{"tcp", "udp"}
```
Comment (remicalixte) [inline pkg/collector/corechecks/net/network.go]: This is essentially a constant, we could define this slice as a toplevel var

**Review by hush-hush — DISMISSED** [review]

**Comment by arbll** [comment]
Commits after 3464d2b are there to be able to specify if a given type of agent (regular vs puppy) should import the config for the go version of a corecheck or for the python one.

File: pkg/collector/corechecks/net/network.go [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+			counters[counterNames[j]] = value
+		}
+	}
+
+	return counters, nil
+}
+
+// Configure configures the network checks
+func (c *NetworkCheck) Configure(rawInstance integration.Data, rawInitConfig integration.Data) error {
+	err := yaml.Unmarshal(rawInitConfig, &c.config.initConf)
```
Comment (xvello) [inline pkg/collector/corechecks/net/network.go]: Can you please call `CheckBase.CommonConfigure` to handle the common options, see https://github.com/DataDog/datadog-agent/blob/master/pkg/collector/corechecks/checkbase.go#L65

**Review by hush-hush — APPROVED** [review]

**Review by derekwbrown — APPROVED** [review]
I built and installed on android 👍

**Pull request merged**